src/Core/Grid/Definition/Factory/AbstractGridDefinitionFactory.php

```diff
@@ -43,7 +44,37 @@
  */
 abstract class AbstractGridDefinitionFactory implements GridDefinitionFactoryInterface
 {
-    use TranslatorAwareTrait, HookDispatcherAwareTrait;
+    use TranslatorAwareTrait;
+
+    /**
+     * @var HookDispatcherInterface
+     */
+    protected $hookDispatcher;
+
+    /**
+     * @param HookDispatcherInterface|null $hookDispatcher
+     */
+    public function __construct(HookDispatcherInterface $hookDispatcher = null)
+    {
+        if (null === $hookDispatcher) {
+            @trigger_error('The $hookDispatcher parameter should not be null, inject your main HookDispatcherInterface service, or NullDispatcher if you don\'t need hooks.', E_USER_DEPRECATED);
+        }
+        $this->hookDispatcher = $hookDispatcher ? $hookDispatcher : new NullDispatcher();
+    }
+
+    /**
+     * Set hook dispatcher.
+     *
+     * @param HookDispatcherInterface $hookDispatcher
+     *
+     * @deprecated
+     */
+    final public function setHookDispatcher(HookDispatcherInterface $hookDispatcher)
```

Comment:
warning; do we type hint on GridDefinitionFactoryInterface somewhere? You need to replace it by the abstract class

Reply:
I don't understand what you mean?